transaction/src/manifest/parser.rs

```diff
@@ -345,6 +353,8 @@ impl SborValueKindIdent {
             "Decimal" => SborValueKindIdent::Decimal,
             "PreciseDecimal" => SborValueKindIdent::PreciseDecimal,
             "NonFungibleLocalId" => SborValueKindIdent::NonFungibleLocalId,
+            "AddressReservation" => SborValueKindIdent::AddressReservation,
+            "NamedAddress" => SborValueKindIdent::NamedAddress,
```

Comment:
So... thinking about this more... I think we have a subtly wrong abstraction.

Really you should be able to have a `Array<Address>(Address("blah"), NamedAddress("blah"))`... which actually suggests to me that `NamedAddress` should actually be the same `ManifestValueKind` as `Address`...

And that basically, `ManifestAddress` is an enum of `Static` or `Named`?

This would simplify some things too... But I appreciate it's quite a big further change. What do you think?

Reply:
As said over slack, will give this a try.